Tests/Observable+MoyaSpec.swift

```diff
@@ -266,5 +266,103 @@ class ObservableMoyaSpec: QuickSpec {
+                expect(receivedObjects?.count) == 3
+                expect(receivedObjects?.map { $0.title }) == ["Hello, Moya!", "Hello, Moya!", "Hello, Moya!"]
+            }
+
+            it("maps data representing a json at a key path to a decodable object") {
+                let json: [String: Any] = ["issue": json] // nested json
+                guard let data = try? JSONSerialization.data(withJSONObject: json, options: .prettyPrinted) else {
+                    preconditionFailure("Failed creating Data from JSON dictionary")
+                }
+                let observable = Response(statusCode: 200, data: data).asObservable()
+
+                var receivedObject: Issue?
+                _ = observable.map(Issue.self, atKeyPath: "issue", using: decoder).subscribe(onNext: { object in
+                    receivedObject = object
+                })
+                expect(receivedObject).notTo(beNil())
+                expect(receivedObject?.title) == "Hello, Moya!"
+                expect(receivedObject?.createdAt) == formatter.date(from: "1995-01-14T12:34:56")!
+            }
+
+            it("ignores invalid data") {
+                var json = json
+                json["createdAt"] = "Hahaha" // invalid date string
+                guard let data = try? JSONSerialization.data(withJSONObject: json, options: .prettyPrinted) else {
+                    preconditionFailure("Failed creating Data from JSON dictionary")
+                }
+                let observable = Response(statusCode: 200, data: data).asObservable()
+
+                var receivedError: Error?
+                _ = observable.map(Issue.self, using: decoder).subscribe { event in
+                    switch event {
+                    case .next:
+                        fail("next called for invalid data")
+                    case .error(let error):
+                        receivedError = error
+                    default:
+                        break
+                    }
+                }
+
+                // check whether we properly got a MoyaError
+                expect(receivedError).to(Predicate<Error> { expression in
+                    let receivedError = try expression.evaluate()
+                    if case MoyaError.objectMapping? = receivedError {
+                        return PredicateResult(status: .matches, message: .expectedActualValueTo("MoyaError.objectMapping"))
+                    } else {
+                        return PredicateResult(status: .doesNotMatch, message: .expectedActualValueTo("MoyaError.objectMapping"))
+                    }
+                })
+
+                // check whether the MoyaError has a DecodingError as an associated value
+                if case let MoyaError.objectMapping(decodingError, _)? = receivedError {
+                    expect(decodingError).to(beAKindOf(DecodingError.self))
+                } else {
+                    preconditionFailure()
```

Comment:
Hmm, is it a `preconditionFailure` though? For me it is a `fail()` as if the test failed, but I'm curious what's your opinion on that one. Also, it is a shame that we cannot avoid this `if` & `preconditionFailure`/`fail` 👿

Reply:
This code will be merged into a single `if-case-let` statement :)